state/state.go

```diff
@@ -1208,17 +1216,9 @@ func (st *State) AddApplication(args AddApplicationArgs) (_ *Application, err er
 	}
 
 	statusDoc := statusDoc{
-		ModelUUID:  st.ModelUUID(),
-		Status:     status.Waiting,
-		StatusInfo: status.MessageWaitForMachine,
-		Updated:    st.clock().Now().UnixNano(),
-		// This exists to preserve questionable unit-aggregation behaviour
-		// while we work out how to switch to an implementation that makes
-		// sense.
-		NeverSet: true,
-	}
-	if model.Type() == ModelTypeCAAS {
-		statusDoc.StatusInfo = status.MessageWaitForContainer
+		ModelUUID: st.ModelUUID(),
+		Status:    status.Unset,
```

Comment:
So now, we won't see `waiting for machine` and `waiting for container` message anymore?

Reply:
not sure if any users have script expects the waiting status

Reply:
If the value is unset, then it will get the information from the units. The unit statuses haven't changed, so they are likely to say "waiting for container". If there are no units, then it will be "unknown"